chaosgcp/apphub/actions.py

```
@@ -0,0 +1,270 @@
+]
+logger = logging.getLogger("chaostoolkit")
+
+
+def get_project_number(project_id: str):
+    """Retrieves the project number for the specified Google Cloud Project ID."""
+    client = resourcemanager_v3.ProjectsClient()
+    # Construct the project name (required format)
+    project_name = f"projects/{project_id}"
+    try:
+        project = client.get_project(name=project_name)
+        return project.name.split("/")[-1]
+        # Extract number from 'projects/123456789'
+    except Exception as e:
+        logger.error(f"Error retrieving project number: {e}")
+        return None
+
+
+def apphub_app_list_services(project_id: str, location: str, application_id: str):
+    """Retrieves a list of service URIs associated with an AppHub application.
+
+    This function queries the AppHub API to list all services deployed within
+    a specific application and extracts their associated URIs. It replaces the
+    project number with the project ID in the URI for improved clarity.
+
+    Args:
+        project_id (str): The ID of the Google Cloud project containing the application.
+        location (str): The geographic location (e.g., "us-central1") of the application.
+        application_id (str): The unique identifier for the AppHub application.
+
+    Returns:
+        list: A list of strings, where each string is the URI of a service associated
+              with the specified application. The URIs are formatted to contain the
+              project ID instead of the project number."""
+    name_app = f"projects/{project_id}/locations/{location}/applications/{application_id}"
+    client = apphub_v1.AppHubClient()
+
+    request = apphub_v1.ListServicesRequest(parent=name_app)
+
+    service_uris = []
+    logger.info("Services for given Application in Apphub")
+    for response in client.list_services(request=request):
+        uri = response.service_reference.uri.replace(
+            get_project_number(project_id), project_id
+        )
```

as get_project_number can return None, this would fail as well. This is why it might be better to raise an error?

---

Handled with try catch